File: 0717-1-bit-and-2-bit-characters/0717-1-bit-and-2-bit-characters.js
/**
 * @param {number[]} bits
 * @return {boolean}
 */
var isOneBitCharacter = function(bits) {
    let i =0
    let res=[]
    while(i<bits.length)
    {
        if(bits[i]==0){
            res.push(0)
        }else if(bits[i]==1 && bits[i+1]!==undefined ){
            res.push([bits[i],bits[i+1]])
            i++
        }
        i++
    }
    // console.log(res)
    if(res[res.length-1]===0){
        return true 
    }
    return false 
};

File: 0717-1-bit-and-2-bit-characters/0717-1-bit-and-2-bit-characters.go
func isOneBitCharacter(bits []int) bool {
    i:=0
    res:=[][]int{}

    for i<len(bits){
        if bits[i]==0{
            res=append(res,[]int{bits[i]})
        }else if bits[i]==1 {
            res=append(res,[]int{bits[i],bits[i+1]})
            i++
        }
        i++
    }
    fmt.Println(res)
    if len(res[len(res)-1])==1{
        return true 
    }
    return false  
}